UMN-AutoPackager/public/Set-UMNGlobalConfig.ps1

```diff
@@ -0,0 +1,153 @@
+function Set-UMNGlobalConfig {
+    <#
+    .SYNOPSIS
+        Sets a value(s) for the global configuration JSON file for the auto packager.
+    .DESCRIPTION
+        This command sets in a json file a value for one of various global configurations used as part of the AutoPackager. It requires a the full path and name of the JSON file, the key to set, and the value to set it to.
+    .EXAMPLE
+        Set-UMNGlobalConfig -Path .\config.json -Key CompanyName  -Value "University of Minnesota"
+        Sets the value of CompanyName to University of Minnesota for the config.json file
+    .EXAMPLE
+        Set-UMNGlobalConfig -Path .\config.json -SiteServer "my.config.site"  -SiteCode "COM" -DownloadLocationPath "C:\\Temp" -ApplicatoinContentPath "\\appstorage.somewhere\"
+        Sets the values for a ConfigMgr site to be added to the config.json file
+    .PARAMETER Path
+        The full path and file name of the JSON file to be updated
+    .PARAMETER Key
+        The key that you want to update in the Global Configuation JSON file
+    .PARAMETER Value
+        The value you want to set the key to
+    .PARAMETER SiteServer
+        The ConfigMgr site server address
+    .PARAMETER SiteCode
+        The site code for this instance of ConfigMgr
+    .PARAMETER DownloadLocationPath
+        The full path where downloaded data will be temporarily stored
+    .PARAMETER ApplicationContentPath
+        The full path for where ConfigMgr applcation content will be stored"
+    #>
+    [CmdletBinding(DefaultParameterSetName="Main")]
+    param (
+        [Parameter(ParameterSetName="Main",Mandatory=$True, HelpMessage = "The full path and file name of the JSON file to be updated")]
+        [Parameter(ParameterSetName="ConfigMgr")]
+        [ValidateNotNullOrEmpty()]
+        [string]$Path,
+        [Parameter(ParameterSetName="Main",Mandatory=$True, HelpMessage = "The key that you want to update in the Global Configuation JSON file")]
+        [ValidateNotNullOrEmpty()]
+        [string]$Key,
+        [Parameter(ParameterSetName="Main",Mandatory=$True, HelpMessage = "The value you want to set the key to")]
+        [ValidateNotNullOrEmpty()]
+        [string]$Value,
+        [Parameter(ParameterSetName="ConfigMgr",Mandatory=$True, HelpMessage = "The ConfigMgr site server address")]
+        [string]$SiteServer,
+        [Parameter(ParameterSetName="ConfigMgr",Mandatory=$True, HelpMessage = "The site code for this instance of ConfigMgr")]
+        [string]$SiteCode,
+        [Parameter(ParameterSetName="ConfigMgr",Mandatory=$True, HelpMessage = "The full path where downloaded data will be temporarily stored")]
+        [string]$DownloadLocationPath,
+        [Parameter(ParameterSetName="ConfigMgr",Mandatory=$True, HelpMessage = "The full path for where ConfigMgr applcation content will be stored")]
+        [string]$ApplicationContentPath
+    )
+    begin {
+        Write-Verbose -Message "Starting $($myinvocation.mycommand) Updating $key with $value"
```

Comment:
As Jeff mentioned on one of my scripts, there are a lot of Write-Verbose lines. Even when running in non-Verbose mode, these still process and output to void. To save on processing power, they should be used where necessary, rather then where they are nice to have. I did the same thing on mine and pared them down a bit.

Reply:
Commented out some of the write-verbose to reduce the number.

Reply:
Thanks! Resolving.